### ISSUE TITLE
'Path' type for commands lib Arguments

### ISSUE THREAD

**Issue by whyrusleeping** [ipfs/go-ipfs#2721]
I've wanted a `PathArg` type for the commands libs that will validate that the content its passed is a valid ipfs path, and allow typed access to that to the consumers of the `cmds.Request`

**Comment by whyrusleeping** [comment]
This would help prevent problems like #2715

**Comment by Kubuxu** [comment]
`valid ipfs path` means resolve it locally?

**Comment by whyrusleeping** [comment]
Not resolve locally, but call `path.ParsePath` on it.

**Comment by Kubuxu** [comment]
Making the check during parsing should be easy. Typed access would be harder and will require changes in most commands definitions.

**Comment by whyrusleeping** [comment]
Nah, i'm thinking that we add a `req.Paths()` accessor or something to the request object. This would require changing a lot of command definitions, but would also clean them up dramatically

**Comment by djdv** [comment]
Is this pull relevant? https://github.com/ipfs/go-ipfs/pull/1740
